docs: Implement the concept of 'common-props' to reduce redundancy in component references

dwgray opened the pull request:
# Describe the PR

This addresses #1957. I pulled the idea from BSV, where they keep a flat file of prop definitions that are re-used in multiple. components.

- @VividLemon reworked the props field to be in object format and type-checked.
- He also added the ability to incorporate 'extensions' in a way that pulls in the property from a different component and puts those definitions in a sub-table (this is demonstrated in `table.md`
- I used `BFormSelect*`, `BFormCheckbox*` and `BFormRadiobutton*` as a trial, it reduced the component reference footprint significantly (although the `BForm*` compnenets are probably the best case for this).

## Small replication

See the `BFormSelect*`, `BFormCheckbox*`, `BFormRadiobutton*` and `BTable` documentation

## PR checklist

<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [x] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [x] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

dwgray commented on apps/docs/src/docs/components/form-radio.md:
```diff
@@ -734,9 +734,10 @@ Supported `aria-invalid` values are:
 `aria-invalid` is automatically set to `true` if the `state` prop is `false`.
 
-<ComponentReference :data="data" />
+<ComponentReference :data="data" :common-props="commonProps" />
```
I'm think I may be able to inject this object from a root component so that I don't have to add this property and the import to every *.md file, but I'd like to get feedback on the approach before diving into that.

VividLemon commented on apps/docs/src/data/components/formCheckbox.data.ts:
```diff
@@ -206,121 +133,58 @@ export default {
-          default: 'value',
-          description: 'Field name in the `options` array that should be used for the value',
-        },
+        {prop: 'valueField'},
```
Rather than this, I'd actually like a schema more like the following

```ts
const commonProps = [
  // Props definitions here
]

// In different file
import {commonProps} from '../'
export default {
load: (): ComponentReference[] => [
  // Previous data
  ...pick(commonProps, ['disabled']) // A pick function exists in the main package I think. It's essentially a TS equivalent to Pick<>
]
```

Then there's also extensions. 

```ts
// For BOverlayProps extends RadiusElementExtendables example
const RadiusElementExtendables = [] 

export default {
  load: () => [
    RadiusElementExtendables
  ]
}
```
Then RadiusElementExtendables appears as a table subsection. 

```md
| Main props          |
|--------------|
| bar          |
| buzz         |
| RadiusElementExtendables  |
|--------------|
| Section Prop |
```

VividLemon replied:
Load looks like `load: (): (ComponentReference | ComponentReference[])[] => [` 
and pick exists in `object.ts`

VividLemon replied:
Alternatively the commonProps can exist as an extension like so, and would not need to be spreaded
```ts
load: () => [
  pick(commonProps, ['disabled'])
]
```

VividLemon replied:
(or if BTable can't do this, multiple concurrent tables would be the same)
```md
| RadiusElementExtendables |
|---------------------|
| bar                 |
| buzz                |

| RadiusElementExtendables |
|---------------------|
| Section Prop        |
```

dwgray replied:
Thanks, @VividLemon - I think I've got the gist of what you're proposing. I'll turn around something based on that next time I've got some time to play with this (probably this week).

VividLemon replied:
A large undertaking for sure, especially [`https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1957#issuecomment-2177246378](https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1957#issuecomment-2177246378)

VividLemon replied:
Alternatively a discord call could clear up any confusion if you had any

VividLemon commented:
@dwgray That should be all the changes I planned on making. You should just need to fill in the blanks

dwgray commented:
@VividLemon thanks for working through this. I agree that the object form reads better for properties and the type-checking actually caught one error in my original work. I updated the *.data.md files that I touched, fixed some build/linting issues in a couple of other files and updated common props.

I think this is ready to go and will unblock me to spend part of my time verifying and filling out the component data for other components as I work through the parity validation. I'm finding this to be slow, but I might as well do all of the validation and get the docs/component data up-to-date all at the same time.

Out of curiosity, did you find a clean way to update the props in the *.data.ts files from array to object format? I got github copilot to do that on a per file basis, but didn't find anything that would let me do a global search and replace equivalent...

VividLemon commented:
fixes  #1584
